app/controllers/application_controller.rb

```diff
@@ -112,8 +112,11 @@ def meursing_lookup_result
   protected
 
   def maintenance_mode_if_active
-    if ENV['MAINTENANCE'].present? && action_name != 'maintenance'
-      redirect_to '/503'
+    if ENV['MAINTENANCE'].present? &&
+        (ENV['MAINTENANCE_BYPASS'].blank? ||
+         params[:maintenance_bypass].blank? ||
+         ENV['MAINTENANCE_BYPASS'] != params[:maintenance_bypass])
```

Comment:
it's pretty difficult to follow this logical expression,

is it the same as this?
```
        !(ENV['MAINTENANCE_BYPASS'].present? && params[:maintenance_bypass].present? &&
         ENV['MAINTENANCE_BYPASS']  == params[:maintenance_bypass])
```

not sure, we should check that maintenance_bypass is also True, and not just present 🤔